testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HydraIdentityProvider.java

```diff
@@ -47,13 +53,18 @@ public void extendEnvironment(Environment.Builder builder)
         DockerContainer hydra = new DockerContainer(HYDRA_IMAGE, "hydra")
                 .withEnv("LOG_LEAK_SENSITIVE_VALUES", "true")
                 .withEnv("DSN", "memory")
-                .withEnv("URLS_SELF_ISSUER", "http://hydra:4444/")
+                .withEnv("URLS_SELF_ISSUER", "https://hydra:4444/")
                 .withEnv("URLS_CONSENT", "http://hydra-consent:3000/consent")
                 .withEnv("URLS_LOGIN", "http://hydra-consent:3000/login")
+                .withEnv("SERVE_TLS_KEY_PATH", "/tmp/certs/localhost.pem")
+                .withEnv("SERVE_TLS_CERT_PATH", "/tmp/certs/localhost.pem")
                 .withEnv("STRATEGIES_ACCESS_TOKEN", "jwt")
                 .withEnv("TTL_ACCESS_TOKEN", TTL_ACCESS_TOKEN_IN_SECONDS + "s")
-                .withCommand("serve all --dangerous-force-http")
-                .waitingFor(Wait.forHttp("/health/ready").forPort(4444).forStatusCode(200));
+                .withCommand("serve", "all")
+                .withCopyFileToContainer(forHostPath(configDir.getPath("cert")), "/tmp/certs")
+                .waitingFor(new WaitAllStrategy()
```

Comment:
Do we need to change waitingFor strategy? The old one should do fine, after being adjusted to Wait.forHttps, isn't it? I'm a bit anxious doing this, as it might potentially affect stability of an environment startup.

Reply:
That's a good point but waiting for HTTPS seems not working:
```
2021/06/01 20:08:23 http: TLS handshake error from 172.20.0.1:63028: remote error: tls: unknown certificate
2021-06-01T22:08:24.120+0200	WARN	main	io.trino.tests.product.launcher.env.Environment	Could not start environment 'singlenode-oauth2'
java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed
	at io.trino.tests.product.launcher.env.Environment.tryStart(Environment.java:157)
	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:48)
....
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for URL to be accessible (https://localhost:4444/health/ready should return HTTP [200])
	at org.testcontainers.containers.wait.strategy.HttpWaitStrategy.waitUntilReady(HttpWaitStrategy.java:264)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:35)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:892)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:440)
	... 9 more
```